x/ecocredit/client/testsuite/tx.go

```diff
@@ -0,0 +1,1096 @@
+			expectedErrMsg: "required flag(s) \"from\" not set",
+		},
+		{
+			name: "single issuer",
+			args: append(
+				[]string{
+					val0.Address.String(),
+					val0.Address.String(),
+					validMetadata,
+					makeFlagFrom(val0.Address.String()),
+				},
+				s.commonTxFlags()...,
+			),
+			expectErr:      false,
+			expectedErrMsg: "",
+			expectedClassInfo: &ecocredit.ClassInfo{
+				Designer: val0.Address.String(),
+				Issuers:  []string{val0.Address.String()},
+				Metadata: []byte{0x1},
+			},
+		},
+		{
+			name: "multiple issuers",
+			args: append(
+				[]string{
+					val0.Address.String(),
+					strings.Join(
+						[]string{
+							val0.Address.String(),
+							val1.Address.String(),
+						},
+						",",
+					),
+					validMetadata,
+					makeFlagFrom(val0.Address.String()),
+				},
+				s.commonTxFlags()...,
+			),
+			expectErr:      false,
+			expectedErrMsg: "",
+			expectedClassInfo: &ecocredit.ClassInfo{
+				Designer: val0.Address.String(),
+				Issuers:  []string{val0.Address.String(), val1.Address.String()},
+				Metadata: []byte{0x1},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			// Commands may panic, so we need to recover and check the error messages
+			defer func() {
+				if r := recover(); r != nil {
+					s.Require().True(tc.expectErr)
+					s.Require().Contains(r.(error).Error(), tc.expectedErrMsg)
+				}
+			}()
+
+			cmd := client.TxCreateClassCmd()
+			out, err := cli.ExecTestCLICmd(clientCtx, cmd, tc.args)
+			if tc.expectErr {
+				s.Require().Error(err)
+				s.Require().Contains(out.String(), tc.expectedErrMsg)
+			} else {
+				s.Require().NoError(err, out.String())
+
+				var res sdk.TxResponse
+				s.Require().NoError(clientCtx.Codec.UnmarshalJSON(out.Bytes(), &res))
+
+				classIdFound := false
+				for _, e := range res.Logs[0].Events {
+					if e.Type == proto.MessageName(&ecocredit.EventCreateClass{}) {
+						for _, attr := range e.Attributes {
+							if attr.Key == "class_id" {
+								classIdFound = true
+								classId := strings.Trim(attr.Value, "\"")
+
+								queryCmd := client.QueryClassInfoCmd()
+								queryArgs := []string{classId, flagOutputJSON}
+								queryOut, err := cli.ExecTestCLICmd(clientCtx, queryCmd, queryArgs)
+								s.Require().NoError(err, queryOut.String())
+								var queryRes ecocredit.QueryClassInfoResponse
+								s.Require().NoError(clientCtx.Codec.UnmarshalJSON(queryOut.Bytes(), &queryRes))
+
+								s.Require().Equal(tc.expectedClassInfo.Designer, queryRes.Info.Designer)
+								s.Require().Equal(tc.expectedClassInfo.Issuers, queryRes.Info.Issuers)
+								s.Require().Equal(tc.expectedClassInfo.Metadata, queryRes.Info.Metadata)
+							}
+						}
+					}
+				}
+				s.Require().True(classIdFound)
+			}
+		})
+	}
+}
+
+// Write MsgCreateBatch to a temporary JSON file and panic on any errors
+func mustWriteBatchJSON(clientCtx sdkclient.Context, msgCreateBatch *ecocredit.MsgCreateBatch) string {
```

Comment:
why not using `WriteToNewTempFile` from cosmos-sdk?
I believe we could also avoid all these panics